app/src/main/java/org/thunderdog/challegram/navigation/ComplexHeaderView.java

```diff
@@ -852,21 +865,33 @@ protected void onDraw (Canvas c) {
         float iconTop = trimmedTitleExpanded != null ? (trimmedTitleExpanded.getHeight() - trimmedTitle.getHeight()) * avatarExpandFactor : 0;
         float iconsAdded = 0;
 
-        if (showVerify) {
-          Paint paint = Paints.getPorterDuffPaint(titleColor);
-          int alpha = paint.getAlpha();
-          paint.setAlpha((int) ((float) alpha * (VERIFY_ALPHA + (1f - VERIFY_ALPHA) * avatarExpandFactor)));
-          Drawable drawable = getSparseDrawable(R.drawable.deproko_baseline_verify_24, 0);
-          Drawables.draw(c, drawable, iconLeft, iconTop + trimmedTitle.getHeight() / 2f - drawable.getMinimumHeight() / 2f, paint);
-          paint.setAlpha(alpha);
-          iconsAdded += drawable.getMinimumWidth();
-        }
+        if (this.subtitle != null && !this.subtitle.equals(Lang.lowercase(Lang.getString(R.string.ChatWithYourself)))) {
```

Comment:
These methods are too heavy for the `onDraw` method. The condition itself seems to be incorrect, as any contact (not only saved messages chat) with name "Saved Messages" would be affected as well. There should be some `FLAG_IS_SELF` instead.

, instead of comparing the strings).